index.js

```diff
@@ -0,0 +1,221 @@
+            this.breaker.runCommand(cb => container.start(cb),
+                err => (err ? reject(err) : resolve())
+            );
+        });
+    }
+
+    /**
+     * Remove a Docker container
+     * @method _removeContainer
+     * @param  {Container}   container Docker container to remove
+     * @return {Promise}
+     */
+    _removeContainer(container) {
+        return new Promise((resolve, reject) => {
+            this.breaker.runCommand(cb => container.remove({ v: true, force: true }, cb),
+                err => (err ? reject(err) : resolve())
+            );
+        });
+    }
+
+    /**
+     * Find Docker containers
+     * @method _findContainers
+     * @param  {String}   buildId Build ID to find
+     * @return {Promise}          List of containers
+     */
+    _findContainers(buildId) {
+        const listArgs = {
+            filters: JSON.stringify({
+                label: [
+                    `sdbuild=${buildId}`
+                ]
+            }),
+            all: true
+        };
+
+        return new Promise((resolve, reject) => {
+            this.breaker.runCommand(cb => this.docker.listContainers(listArgs, cb),
+                (err, containers) => (err ? reject(err) : resolve(containers))
+            );
+        });
+    }
+
+    /**
+     * Starts a s3m build
+     * @method _start
+     * @param  {Object}   config            A configuration object
+     * @param  {String}   config.buildId    ID for the build
+     * @param  {String}   config.container  Container for the build to run in
+     * @param  {String}   config.apiUri     API Uri
+     * @param  {String}   config.token      JWT for the Build
+     * @return {Promise}
+     */
+    _start(config) {
+        return this._createContainer(
+            {
+                name: `${config.buildId}-init`,
+                Image: `screwdrivercd/launcher:${this.launchVersion}`,
+                Entrypoint: '/bin/true',
+                Labels: {
+                    sdbuild: config.buildId
+                }
+            })
+            .then(launchContainer => Promise.all([
+                this._createContainer({
+                    name: `${config.buildId}-log`,
+                    Image: `screwdrivercd/log-service:${this.logVersion}`,
+                    Entrypoint: '/opt/screwdriver/tini',
+                    Labels: {
+                        sdbuild: config.buildId
+                    },
+                    Cmd: [
+                        '--',
+                        '/opt/screwdriver/logservice',
+                        '--emitter',
+                        '/opt/screwdriver/emitter',
+                        config.buildId
+                    ],
+                    Env: [
+                        `SD_TOKEN=${config.token}`
+                    ],
+                    HostConfig: {
+                        Memory: 200 * 1024 * 1024,
+                        MemoryLimit: 300 * 1024 * 1024,
```

Comment:
Might be worth mentioning what units these are in.

As an aside, what's the difference between the two settings?